On my android the app crashes after upgrading to 1.8.7

I believe I'm experiencing the same issue, here are my logs: 

```
A Dart VM Service on Bonophone is available at: http://127.0.0.1:62833/od9H2P7x3TU=/
The Flutter DevTools debugger and profiler on Bonophone is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:62833/od9H2P7x3TU=/
flutter: [WARN] TIME: 2024-02-19T18:28:49.647788Z r: Initializing fee rate cache with default values. defaults: [(MempoolMinimum, 1000), (Background, 2000), (Normal, 3000), (HighPriority, 4000)]
flutter: [INFO] TIME: 2024-02-19T18:28:50.439150Z r: Found channel manager data. Recovering from stored state
flutter: [INFO] TIME: 2024-02-19T18:28:50.632159Z r: Listening on 0.0.0.0:56017
flutter: [INFO] TIME: 2024-02-19T18:28:50.632725Z r: Starting background processor
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.632918Z r: Getting all non-pending channels
flutter: [INFO] TIME: 2024-02-19T18:28:50.638680Z r: Lightning node started with node ID 0260f5d3f3f90551729c3df880be7214be3a0dcf7f7cc421f44cac0938554fca80@0.0.0.0:56017
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.644261Z r: Syncing 0 shadow channels
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.644619Z r: Got all transactions transactions: []
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.644688Z r: Syncing 0 shadow transactions
flutter: [INFO] TIME: 2024-02-19T18:28:50.653772Z r: Keeping RGS snapshot up to date
flutter: [TRACE] TIME: 2024-02-19T18:28:50.655044Z r: lightning_background_processor: Calling ChannelManager's timer_tick_occurred on startup
flutter: [INFO] TIME: 2024-02-19T18:28:50.655220Z r: Requesting RGS gossip update latest_sync_timestamp: 1708354800,rgs_server_url: https://rapidsync.lightningdevkit.org/snapshot
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.659746Z r: Got all spendable outputs outputs: []
flutter: [TRACE] TIME: 2024-02-19T18:28:50.665180Z r: lightning_background_processor: Rebroadcasting monitor's pending claims on startup
flutter: [TRACE] TIME: 2024-02-19T18:28:50.799049Z Successfully synced payment history
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.799375Z found no subscribers, skipping event
flutter: [INFO] TIME: 2024-02-19T18:28:50.799922Z r: Started on-chain sync
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.799999Z r: Setting up connection peer: 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156@46.17.98.29:9045
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.800056Z r: Connecting to orderbook API
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.800106Z r: Pruning expired orders
flutter: [ERROR] TIME: 2024-02-19T18:28:50.803077Z Failed to start backend. FfiException(RESULT_ERROR, A failed order must have a reason, null)
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.817325Z r: Response status code 200 OK
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.817496Z r: Successfully uploaded backup of dlc/03/636861696e5f6d6f6e69746f72.
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.839475Z r: Connection setup completed peer: 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156@46.17.98.29:9045
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.839583Z r: Waiting to confirm established connection peer: 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156@46.17.98.29:9045
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.839643Z r: Response status code 200 OK
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.839693Z r: Successfully uploaded backup of dlc/03/636861696e5f6d6f6e69746f72.
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.862586Z r: lightning::ln::peer_handler: Finished noise handshake for connection with 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.863120Z r: tungstenite::handshake::client: Client handshake done.
flutter: [INFO] TIME: 2024-02-19T18:28:50.865152Z r: Connected to orderbook realtime API
flutter: [TRACE] TIME: 2024-02-19T18:28:50.900698Z r: lightning_rapid_gossip_sync::processing: Processing RGS data...
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.901596Z r: lightning_rapid_gossip_sync::processing: Processing RGS update from 1708354800 with 0 nodes, 0 channel announcements and 0 channel updates.
flutter: [INFO] TIME: 2024-02-19T18:28:50.901687Z r: Updated network graph new_latest_sync_timestamp: 1708354800
flutter: [TRACE] TIME: 2024-02-19T18:28:50.921928Z r: lightning::ln::channelmanager: New best block: 00000000000000000002e19243e556eccff322dd5600629f39bf75822c44788c at height 831152
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.922141Z r: lightning::chain::chainmonitor: New best block 00000000000000000002e19243e556eccff322dd5600629f39bf75822c44788c at height 831152 provided via best_block_updated
flutter: [TRACE] TIME: 2024-02-19T18:28:50.951764Z r: lightning_background_processor: Persisting ChannelManager...
flutter: [TRACE] TIME: 2024-02-19T18:28:50.996040Z r: lightning_background_processor: Done persisting ChannelManager.
flutter: [INFO] TIME: 2024-02-19T18:28:50.996465Z r: lightning::ln::peer_handler: Received peer Init message from 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156: DataLossProtect: required, InitialRoutingSync: not supported, UpfrontShutdownScript: supported, GossipQueries: not supported, VariableLengthOnion: required, StaticRemoteKey: required, PaymentSecret: required, BasicMPP: supported, Wumbo: supported, AnchorsNonzeroFeeHtlcTx: not supported, AnchorsZeroFeeHtlcTx: not supported, ShutdownAnySegwit: supported, Taproot: not supported, OnionMessages: not supported, ChannelType: supported, SCIDPrivacy: supported, ZeroConf: supported, unknown flags: none
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.997010Z r: lightning::ln::channelmanager: Generating channel_reestablish events for 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156
flutter: [INFO] TIME: 2024-02-19T18:28:50.997120Z r: Successfully logged in to 10101 websocket api!
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.998955Z r: Received all orders from orderbook orders: []
flutter: [INFO] TIME: 2024-02-19T18:28:50.999037Z r: Peer connected! inbound: false,their_node_id: 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156
flutter: [DEBUG] TIME: 2024-02-19T18:28:50.999102Z r: Updating prices prices: {BtcUsd: Price { bid: Some(52099.5), ask: Some(52100) }}
flutter: [INFO] TIME: 2024-02-19T18:28:51.002142Z r: Processing message from: 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156,kind: ChannelSettleConfirm
flutter: [DEBUG] TIME: 2024-02-19T18:28:51.007101Z r: Sending last dlc message last_outbound_serialized_dlc_message.message_type: SettleFinalize,peer: 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156
flutter: [DEBUG] TIME: 2024-02-19T18:28:51.016871Z r: Received message that has already been processed, skipping. kind: ChannelSettleConfirm,node_id: 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156
flutter: [INFO] TIME: 2024-02-19T18:28:51.830743Z r: Connection established peer: 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156@46.17.98.29:9045
flutter: [INFO] TIME: 2024-02-19T18:28:55.275714Z r: Finished on-chain sync duration: 4480,latest_height: 831152
flutter: [TRACE] TIME: 2024-02-19T18:28:55.955569Z Successfully synced payment history
flutter: [TRACE] TIME: 2024-02-19T18:29:01.024385Z Successfully synced payment history
flutter: [TRACE] TIME: 2024-02-19T18:29:06.115387Z Successfully synced payment history
flutter: [TRACE] TIME: 2024-02-19T18:29:11.215178Z Successfully synced payment history
flutter: [WARN] TIME: 2024-02-19T18:29:14.387395Z r: Could not remove non-existing order order_id: 22a078f3-db74-4f12-bd04-c966fad443e3
flutter: [DEBUG] TIME: 2024-02-19T18:29:14.388622Z r: Updating prices prices: {BtcUsd: Price { bid: Some(52101), ask: Some(52100) }}
flutter: [DEBUG] TIME: 2024-02-19T18:29:16.473178Z r: Updating prices prices: {BtcUsd: Price { bid: Some(52101), ask: Some(52101.5) }}
flutter: [TRACE] TIME: 2024-02-19T18:29:16.479396Z Successfully synced payment history
flutter: [DEBUG] TIME: 2024-02-19T18:29:20.800911Z r: Pruning expired orders
flutter: [DEBUG] TIME: 2024-02-19T18:29:20.815384Z r: Response status code 200 OK
flutter: [DEBUG] TIME: 2024-02-19T18:29:20.815949Z r: Successfully uploaded backup of dlc/03/636861696e5f6d6f6e69746f72.
flutter: [DEBUG] TIME: 2024-02-19T18:29:20.817945Z r: Response status code 200 OK
flutter: [DEBUG] TIME: 2024-02-19T18:29:20.818501Z r: Successfully uploaded backup of dlc/03/636861696e5f6d6f6e69746f72.
flutter: [TRACE] TIME: 2024-02-19T18:29:20.870266Z r: lightning_background_processor: Calling PeerManager's timer_tick_occurred
flutter: [TRACE] TIME: 2024-02-19T18:29:20.871221Z r: lightning_background_processor: Rebroadcasting monitor's pending claims
flutter: [TRACE] TIME: 2024-02-19T18:29:21.675623Z Successfully synced payment history
```

This seems to be the cause  : 

```
flutter: [ERROR] TIME: 2024-02-19T18:28:50.803077Z Failed to start backend. FfiException(RESULT_ERROR, A failed order must have a reason, null)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+- Fix(mobile): Fixed a bug where the app didn't load. This was because of some inconsistency in the db where a failed order didn't have a failure reason.
+
 ## [1.8.9] - 2024-02-19
 
 - Fix(validation): Show correct max counterparty collateral in validation message.

--- a/mobile/lib/features/welcome/loading_screen.dart
+++ b/mobile/lib/features/welcome/loading_screen.dart
@@ -79,7 +79,7 @@ class _LoadingScreenState extends State<LoadingScreen> {
       }
     }).catchError((error) {
       logger.e("Failed to start backend. $error");
-      showSnackBar(ScaffoldMessenger.of(context), "Failed to start 10101!");
+      showSnackBar(ScaffoldMessenger.of(context), "Failed to start 10101! $error");
     });
   }
 

--- a/mobile/native/src/db/models.rs
+++ b/mobile/native/src/db/models.rs
@@ -47,8 +47,6 @@ pub enum Error {
     MissingPriceForLimitOrder,
     #[error("A filling or filled order has to have an execution price")]
     MissingExecutionPrice,
-    #[error("A failed order must have a reason")]
-    MissingFailureReason,
 }
 
 #[derive(Queryable, QueryableByName, Insertable, Debug, Clone, PartialEq)]
@@ -88,7 +86,10 @@ impl Order {
     pub fn set_all_filling_orders_to_failed(conn: &mut SqliteConnection) -> Result<()> {
         let affected_rows = diesel::update(orders::table)
             .filter(schema::orders::state.eq(OrderState::Filling))
-            .set(orders::state.eq(OrderState::Failed))
+            .set((
+                orders::state.eq(OrderState::Failed),
+                orders::failure_reason.eq(FailureReason::Unknown),
+            ))
             .execute(conn)?;
 
         tracing::info!("Updated {affected_rows} orders from Filling to Failed");
@@ -625,7 +626,9 @@ impl TryFrom<(OrderState, Option<f32>, Option<FailureReason>)> for crate::trade:
             OrderState::Rejected => crate::trade::order::OrderState::Rejected,
             OrderState::Open => crate::trade::order::OrderState::Open,
             OrderState::Failed => match value.2 {
-                None => return Err(Error::MissingFailureReason),
+                None => crate::trade::order::OrderState::Failed {
+                    reason: crate::trade::order::FailureReason::Unknown,
+                },
                 Some(reason) => crate::trade::order::OrderState::Failed {
                     reason: reason.into(),
                 },

